soc/nordic/common/uicr/uicrgen.py

```diff
@@ -0,0 +1,272 @@
+        periphconf_address = flash_base_address + periphconf_partition.regs[0].addr
+        periphconf_size = periphconf_partition.regs[0].size
+
+        periphconf_combined = extract_and_combine_periphconfs(args.in_periphconf_elfs)
+        padding_len = periphconf_size - len(periphconf_combined)
+        periphconf_final = periphconf_combined + bytes([0xFF for _ in range(padding_len)])
+
+        if kconfig["CONFIG_NRF_HALTIUM_UICR_PERIPHCONF"] == "y":
+            uicr.PERIPHCONF.ENABLE = ENABLED_VALUE
+            uicr.PERIPHCONF.ADDRESS = periphconf_address
+            uicr.PERIPHCONF.MAXCOUNT = math.floor(periphconf_size / 8)
+
+        try:
+            uicr_node = edt.label2node[UICR_NODELABEL]
+        except LookupError as e:
+            raise ScriptError(
+                "Failed to find UICR node in the devicetree. "
+                f"Expected a DT node with label '{UICR_NODELABEL}'."
+            ) from e
+
+        uicr_hex = IntelHex()
+        uicr_hex.frombytes(bytes(uicr), offset=uicr_node.regs[0].addr)
+
+        uicr_hex.write_hex_file(args.out_uicr_hex)
+
+        if args.out_periphconf_hex is not None:
+            periphconf_hex = IntelHex()
+            periphconf_hex.frombytes(periphconf_final, offset=periphconf_address)
+            periphconf_hex.write_hex_file(args.out_periphconf_hex)
+
+    except ScriptError as e:
+        print(f"Error: {e!s}")
+        sys.exit(1)
+
+
+def extract_and_combine_periphconfs(elf_files: list[argparse.FileType]) -> bytes:
+    combined_periphconf = []
+
+    for in_file in elf_files:
+        elf = ELFFile(in_file)
+        conf_section = elf.get_section_by_name(PERIPHCONF_SECTION)
+        if conf_section is None:
+            raise ScriptError(f"Failed to find section '{PERIPHCONF_SECTION}' in {in_file.name}")
+
+        conf_section_data = conf_section.data()
+        num_entries = len(conf_section_data) // PERIPHCONF_ENTRY_SIZE
+        periphconf = (PeriphconfEntry * num_entries).from_buffer_copy(conf_section_data)
+        combined_periphconf.extend(periphconf)
+
+    combined_periphconf.sort(key=lambda e: e.regptr)
+    deduplicated_periphconf = []
+
+    for regptr, regptr_entries in groupby(combined_periphconf, key=lambda e: e.regptr):
+        entries = list(regptr_entries)
+        if len(entries) > 1:
+            unique_values = set(e.value for e in entries)
+            if len(unique_values) > 1:
+                raise ScriptError(
+                    f"PERIPHCONF has conflicting values for register 0x{regptr:09_x}: "
+                    + ", ".join([f"0x{val:09_x}" for val in unique_values])
+                )
+        deduplicated_periphconf.append(entries[0])
+
+    final_periphconf = (PeriphconfEntry * len(deduplicated_periphconf))()
+    for i, entry in enumerate(deduplicated_periphconf):
+        final_periphconf[i] = entry
+
+    return bytes(final_periphconf)
+
+
+def parse_kconfig(content: str) -> dict[str, str | None]:
```

Comment:
Hmm is this the first time a .config file gets loaded as a dictionary and there is no utility library somewhere?